src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const STATES = [];
+  let copy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        copy = {};
+        break;
+      case 'addProperties':
+        copy = { ...copy, ...action.extraData };
+        break;
+      case 'removeProperties':
+        copy = { ...copy };
```

Comment:
[CODE STYLE] - There is no need to make a copy of 'copy' again inside the 'removeProperties' case since it's already a separate object from 'state'. You can directly delete the keys from 'copy'.